src/main/java/edu/cornell/kfs/module/purap/document/service/impl/CuB2BShoppingServiceImpl.java

```diff
@@ -183,15 +192,60 @@ public List createRequisitionsFromCxml(B2BShoppingCart message, Person user) thr
 
             req.fixItemReferences();
 
-            // save requisition to database
-            purapService.saveDocumentNoValidation(req);
+            if (checkRequisitionAccountsAreUnique(req)) {
+                // save requisition to database
+                purapService.saveDocumentNoValidation(req);
 
-            // add requisition to List
-            requisitions.add(req);
+                // add requisition to List
+                requisitions.add(req);
+            }
+        }
+
+        if (Collections.isEmpty(requisitions)) {
+            throw new B2BShoppingException("Invalid e-shop Request. No requisitions were created.");
         }
+
         return requisitions;
     }
 
+    public boolean checkRequisitionAccountsAreUnique(RequisitionDocument requisitionDocument) {
```

Comment:
Andrew,

Is this the same validation that occurs when a user manually creates a Requisition in KFS? Is there any way to plug in that validation here so that we can make sure the exact same validation occurs in both cases? Thank you.

Reply:
I believe the validation is in the base code PurchasingAccountsPayableUniqueAccountingStringsValidation.validate. I could injecting that validation bean into the CuB2bShoppingServiceImpl and passing in a dummy AttributedDocumentEvent parameter?

Reply:
That should work.  Looks like the object is already registered in Spring: PurchasingAccountsPayable-uniqueAccountingStringsValidation .  We should switch to use this validation routine as Daniela suggested.